### ISSUE TITLE
Why does the PKCE flow use the org authorization server configuration endpoint?

### ISSUE THREAD

**Issue by colvardrm2** [okta/okta-auth-js#297]
Question and possibly a bug. I'm working on getting an app running with PKCE using the @okta/okta-react npm package, and I noticed that I got a CORS error because it requests the configuration from the org authorization server (https://{oktaDomain}/.well-known/openid-configuration) rather than the actual issuer that is specified in the config (https://{oktaDomain}/oauth2/{authServerId}/.well-known/openid-configuration). I can update the configuration for the server, but is this intentional? It seems strange to request the configuration for a server that isn't the one you're actually using to authenticate, but I might be missing something in the documentation.

From what I can tell, the steps that lead to this are:
1. [Get a url that strips the custom auth server data from the issuer](https://github.com/okta/okta-auth-js/blob/37ec1bc474152fb83088d21500fd5b325b87f2f4/packages/okta-auth-js/lib/browser/browser.js#L35)
2. [If using PKCE, retrieve the configuration from the server](https://github.com/okta/okta-auth-js/blob/3e8c654b99de771549775eb566f9349c86ed89b6/packages/okta-auth-js/lib/token.js#L585). Here, the issuer supplied to [the `getWellKnown` function](https://github.com/okta/okta-auth-js/blob/3e8c654b99de771549775eb566f9349c86ed89b6/packages/okta-auth-js/lib/oauthUtil.js#L81) is null, so it uses the stripped url that points to the org authorization server.

**Comment by swiftone** [comment]
Hey @colvardrm2 - Thanks for the helpfully detailed report.  We'll take a look.

Are you having any immediate functional issue as a result that requires a workaround?

**Comment by swiftone** [comment]
Internal ref: OKTA-269808

**Comment by colvardrm2** [comment]
@swiftone In our particular case, the org authorization server gave a CSP error on attempting to hit the /.well-known/openid-configuration endpoint. We had a workaround to just set the 'url' prop on the Auth object from @okta/okta-react to the same value as the 'issuer' prop to force it to hit the custom authorization server. We didn't see any errors as a result of this change.

Unfortunately, I have limited access to the auth server where the configurations are maintained, so I don't know all of the exact configurations that were changed to fix the CSP error we were encountering. I don't know of any reason why the configuration would be shared between the org server and custom authorization servers, though.
